File: _정답코드/4주차/02_피보나치_수_2/Solution.java
public class Solution {
    private static int count1 = 0;
    private static int count2 = 0;
    public int[] solution(int n) {
        fibo(n);
        int[] answer = {count1, count2};
        return answer;
    }

    private void fibo(int n){
        if(n == 1){
            count1 ++;
        } else if (n == 2){
            count2 ++;
        } else {
            fibo(n-1);
            fibo(n-2);
        }
    }
}

File: _정답코드/4주차/02_피보나치_수_2/solution.py
def solution(n):

    def fibo(n):
        nonlocal count1, count2
        if n == 1:
            count1 += 1
        elif n == 2:
            count2 += 1
        else:
            fibo(n - 1)
            fibo(n - 2)
    
    count1 = 0
    count2 = 0

    fibo(n)
    
    return [count1, count2]

